runtime/libpgmath/lib/common/arm64intrin.h

```diff
@@ -0,0 +1,2826 @@
+  }
+
+  inline operator int32x4_t() const {
+    return nird;
+  }
+
+  inline operator int64x2_t() const {
+    return nlrd;
+  }
+
+  inline __s128i operator+(unsigned int i) const {
+    __s128i r(*this);
+
+    if (i != 0U) {
+      r.xird[0] += i;
+      r.xird[1] += i;
+      r.xird[2] += i;
+      r.xird[3] += i;
+    }
+
+    return r;
+  }
+
+  inline __s128i operator+(int i) const {
+    __s128i r(*this);
+
+    if (i != 0) {
+      r.xird[0] += i;
+      r.xird[1] += i;
+      r.xird[2] += i;
+      r.xird[3] += i;
+    }
+
+    return r;
+  }
+
+  inline __s128i operator+(long l) const {
+    __s128i r(*this);
+
+    if (l != 0L) {
+      r.xlrd[0] += l;
+      r.xlrd[1] += l;
+    }
+
+    return r;
+  }
+
+  inline __s128i operator+(unsigned long l) const {
+    __s128i r(*this);
+
+    if (l != 0UL) {
+      r.xul[0] += l;
+      r.xul[1] += l;
+    }
+
+    return r;
+  }
+
+  inline __s128i operator+(float f) const {
+    __s128i r(*this);
+
+    if (f != 0.0) {
+      r.xird[0] += f;
+      r.xird[1] += f;
+      r.xird[2] += f;
+      r.xird[3] += f;
+    }
+
+    return r;
+  }
+
+  inline __s128i operator+(double d) const {
+    __s128i r(*this);
+
+    if (d != 0.0) {
+      r.xird[0] += d;
+      r.xird[1] += d;
+      r.xird[2] += d;
+      r.xird[3] += d;
+    }
+
+    return r;
+  }
+
+  inline __s128f operator/(const __s128i& rhs) const {
+    __s128f r;
+
+    if (type == rhs.type && type == SignedInt) {
+      r.xfrd[0] = (float) xird[0] / rhs.xird[0];
+      r.xfrd[1] = (float) xird[1] / rhs.xird[1];
+      r.xfrd[2] = (float) xird[2] / rhs.xird[2];
+      r.xfrd[3] = (float) xird[3] / rhs.xird[3];
+    } else if (type == rhs.type && type == UnsignedInt) {
+      r.xfrd[0] = (float) xui[0] / rhs.xui[0];
+      r.xfrd[1] = (float) xui[1] / rhs.xui[1];
+      r.xfrd[2] = (float) xui[2] / rhs.xui[2];
+      r.xfrd[3] = (float) xui[3] / rhs.xui[3];
+    }
+
+    return r;
+  }
+
+  inline __s128f operator/(int i) const {
+    __s128f r;
+
+    r.xfrd[0] = (float) xird[0] / i;
+    r.xfrd[1] = (float) xird[1] / i;
+    r.xfrd[2] = (float) xird[2] / i;
+    r.xfrd[3] = (float) xird[3] / i;
+
+    return r;
+  }
+
+  inline __s128f operator/(unsigned int i) const {
+    __s128f r;
+
+    r.xfrd[0] = (float) xui[0] / i;
+    r.xfrd[1] = (float) xui[1] / i;
+    r.xfrd[2] = (float) xui[2] / i;
+    r.xfrd[3] = (float) xui[3] / i;
+
+    return r;
+  }
+
+  inline __s128i operator+(const __s128i& s) const {
+    __s128i r;
+    r = xird + s.xird;
+    return r;
+  }
+
+  inline __s128i operator-(const __s128i& s) const {
+    __s128i r;
+    r = xird - s.xird;
+    return r;
+  }
+
+  inline __s128i operator_() const {
+    return __s128i(-xird[0], -xird[1], -xird[2], xird[3]);
+  }
+
+  inline __s128i operator+(const long double& ld) const {
+    __s128i r(*this);
+
+    if (ld != 0.0)
+      r.b += ld;
+
+    return r;
+  }
+
+  inline __s128i& operator+=(unsigned int i) {
+    if (i != 0U)
+      xird = xird + i;
```

Comment:
https://github.com/flang-compiler/flang/issues/557
Looks like the clang is complaining about the conversion from unsigned int to int.
From GCC docs:
For convenience, it is allowed to use a binary vector operation where one operand is a scalar. In that case the compiler transforms the scalar operand into a vector where each element is the scalar from the operation. The transformation happens only if the scalar could be safely converted to the vector-element type.